diff --git a/cjs/index.js b/cjs/index.js
index e92a5f6c359b8667cc7da35f1812e80d3becf96c..11cc2fd44b912aa13dd14400b4e2cf5da76a83df 100644
--- a/cjs/index.js
+++ b/cjs/index.js
@@ -5,7 +5,7 @@ exports.transformSchemaAST = transformSchemaAST;
 const path_1 = require("path");
 const graphql_1 = require("graphql");
 const plugin_helpers_1 = require("@graphql-codegen/plugin-helpers");
-const plugin = async (schema, _documents, { commentDescriptions = false, includeDirectives = false, includeIntrospectionTypes = false, sort = false, federation, }) => {
+const plugin = async (schema, _documents, { commentDescriptions = false, includeDirectives = false, includeIntrospectionTypes = false, sort = false, federation, append = '' }) => {
     const transformedSchemaAndAst = transformSchemaAST(schema, {
         sort,
         federation,
@@ -18,6 +18,7 @@ const plugin = async (schema, _documents, { commentDescriptions = false, include
             : graphql_1.printSchema(transformedSchemaAndAst.schema, {
                 commentDescriptions,
             }),
+        append,
     ]
         .filter(Boolean)
         .join('\n');
diff --git a/esm/index.js b/esm/index.js
index f4d86178052166577407a8e6fd55b0d3e8b83e02..ce9d795ac6b4403b094a1795ab28e6cc845a344f 100644
--- a/esm/index.js
+++ b/esm/index.js
@@ -1,7 +1,7 @@
 import { extname } from 'path';
 import { buildASTSchema, extendSchema, parse, print, printIntrospectionSchema, printSchema, visit, } from 'graphql';
 import { getCachedDocumentNodeFromSchema, removeFederation, } from '@graphql-codegen/plugin-helpers';
-export const plugin = async (schema, _documents, { commentDescriptions = false, includeDirectives = false, includeIntrospectionTypes = false, sort = false, federation, }) => {
+export const plugin = async (schema, _documents, { commentDescriptions = false, includeDirectives = false, includeIntrospectionTypes = false, sort = false, federation, append = '' }) => {
     const transformedSchemaAndAst = transformSchemaAST(schema, {
         sort,
         federation,
@@ -14,6 +14,7 @@ export const plugin = async (schema, _documents, { commentDescriptions = false,
             : printSchema(transformedSchemaAndAst.schema, {
                 commentDescriptions,
             }),
+        append,
     ]
         .filter(Boolean)
         .join('\n');
diff --git a/typings/index.d.cts b/typings/index.d.cts
index 935733cce591359ce0345db501d6b58dc4b9c8ef..d57cf3269ad20dd098a89131a2760a1bc6d722d0 100644
--- a/typings/index.d.cts
+++ b/typings/index.d.cts
@@ -27,6 +27,8 @@ export interface SchemaASTConfig {
      * ```
      */
     includeDirectives?: boolean;
+    /** Text to be appended at the end of the generated output. */
+    append?: string;
     /**
      * @description Include introspection types to Schema output.
      * @default false
diff --git a/typings/index.d.ts b/typings/index.d.ts
index 935733cce591359ce0345db501d6b58dc4b9c8ef..d57cf3269ad20dd098a89131a2760a1bc6d722d0 100644
--- a/typings/index.d.ts
+++ b/typings/index.d.ts
@@ -27,6 +27,8 @@ export interface SchemaASTConfig {
      * ```
      */
     includeDirectives?: boolean;
+    /** Text to be appended at the end of the generated output. */
+    append?: string;
     /**
      * @description Include introspection types to Schema output.
      * @default false
